feat(#28): develop accordion, accordion-item and faq-section components

ssibrahimbas opened the pull request:
## What has been developed?

- [x] accordion component
- [x] accordion item component
- [x] faq section component

screenshots

## Desktop

![Screenshot 2023-01-22 at 15 55 53](https://user-images.githubusercontent.com/76786120/213917100-918a1c04-85c8-480c-a3d3-1cf8a7641bdb.png)

## Tablet

![Screenshot 2023-01-22 at 15 55 59](https://user-images.githubusercontent.com/76786120/213917101-2ef0d5f9-6400-4861-a0cb-5e2bbaf4b7b5.png)

## Mobile

![Screenshot 2023-01-22 at 15 56 10](https://user-images.githubusercontent.com/76786120/213917106-b8af04aa-f099-4a66-9047-7f5fcb938cc1.png)

eraygundogmus commented on components/Accordion/AccordionItem.tsx:
```diff
@@ -0,0 +1,28 @@
+      >
+        <div className="grow text-lg font-medium text-gray-900">{title}</div>
+        <div className="text-indigo-500">
+          {isExpanded ? <MinusCircle /> : <PlusCircle />}
```
I think it would be better if you could animate on toggle

ssibrahimbas replied:
I agree. I created a issue for it. 

Please look at #56

eraygundogmus commented on components/Accordion/AccordionItem.tsx:
```diff
@@ -0,0 +1,28 @@
+  isExpanded
+}) => {
+  return (
+    <div className={'border-b py-3'}>
```
why are these curly brackets?

ssibrahimbas replied:
a little hasty mistake, I'm fixed on next commit

eraygundogmus commented on components/Accordion/Accordion.tsx:
```diff
@@ -0,0 +1,39 @@
+import { AccordionProps } from './Accordion.types';
+
+const Accordion: React.FC<AccordionProps> = ({ items }) => {
+  const [openKey, setOpenKey] = React.useState<string | null>(null);
```
is there any spesific reason for you to make it only string? maybe we could keep it as a string array too so we can control it with a prop. 

even if it's not necessary for this design, we can consider we're gonna need it on another project on frontendship, maybe we will use it a UI kit. 

Choice on you!

ssibrahimbas replied:
I actually did it so that when one opens, the other closes. Now I do this through the item. Auto power off is currently disabled.

Thanks :)

oguzergul commented:
Design should be applied precisely the same. Your screenshots show us we have some UI issues like title margins. Accordion item padding, font styles etc.

oguzergul commented on components/Accordion/AccordionItem.tsx:
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
```
I think we use should also use the same Compound Components pattern here. Because the accordion Component is not flexible with this structure. It works for data now. 


`
<Accordion>
        <Accordion.Item> Test Accordion </Accordion.Item> 
        <Accordion.Detail>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
        </Accordion.Details>
</Accordion>
`

oguzergul commented on components/Accordion/Accordion.tsx:
```diff
@@ -0,0 +1,39 @@
+const Accordion: React.FC<AccordionProps> = ({ items }) => {
+  const [openKey, setOpenKey] = React.useState<string | null>(null);
+
+  const updateUrlHash = (key: string | null) => {
```
we don't need to use URL for accordion component.

oguzergul commented on components/Accordion/Accordion.tsx:
```diff
@@ -0,0 +1,39 @@
+  };
+
+  const onToggle = (key: string | null) => {
+    if (openKey === key) key = null;
```
Keyboard keys should be specified as default like "Enter or Space". Don't need to make it extendable.

oguzergul commented on components/Accordion/Accordion.tsx:
```diff
@@ -0,0 +1,39 @@
+    <div className="w-full">
+      {items.map((item, index) => (
+        <AccordionItem
+          title={item.title}
```
Reduce props and solve them with react patterns. Will not be needed anymore these props after using a compound pattern.

The accordion component can include lots of different children. Not only title or content.

ssibrahimbas commented:
What has been done?

- [x] Accordion component has been made more generally available.
- [x] Improved Accordion.Item, Accordion.Details and Accordion.Header components.
- [x] Updated padding and margins to match ui design.
- [x] Removed url hash functionality.
- [x] opened any item instead of a single item.
- [x] Removed tag value from faq item.

What will change according to other issues?

- [ ] An issue has been opened for adding animation. Accordion should use it when this issue is completed.

### Example Images

#### Mobile


![Screenshot 2023-01-23 at 20 52 04](https://user-images.githubusercontent.com/76786120/214113058-91771348-518a-498c-b467-aa574be79ec1.png)


![Screenshot 2023-01-23 at 20 51 10](https://user-images.githubusercontent.com/76786120/214113052-2c1517e1-fad7-4a93-a05d-663e9b5327a5.png)

![Screenshot 2023-01-23 at 20 51 02](https://user-images.githubusercontent.com/76786120/214113038-e7522343-e10c-42be-98f3-a56bfb73fa5c.png)

oguzergul commented on components/FaqSection/FaqSection.tsx:
```diff
@@ -0,0 +1,29 @@
+          Everything you need to know about the product and billing.
+        </p>
+      </div>
+      <Accordion>
```
We should use the `Accordion `as a component, not an `Accordion.Item. `You need to move up to the parent.

oguzergul replied:
The image is not similar to our design. Accordion dividers are not visible, fonts are not the same, and padding is more than the design. Please check your applied CSS codes.

oguzergul commented on components/Accordion/Accordion.tsx:
```diff
@@ -0,0 +1,56 @@
+        className="flex cursor-pointer items-center justify-between"
+      >
+        {children[0]}
+        <div className="text-indigo-500">
```
this component is a button. you need to change this.

oguzergul commented on components/Accordion/Accordion.tsx:
```diff
@@ -0,0 +1,56 @@
+
+  return (
+    <div className={'border-b py-3 md:pb-7'}>
+      <div
```
this is also a button as a semantic mean. you can check this site for semantic and accessible pattern for accordion component. https://www.w3.org/WAI/ARIA/apg/example-index/accordion/accordion.html

oguzergul commented on components/Accordion/Accordion.tsx:
```diff
@@ -0,0 +1,56 @@
+  return <div className="w-full">{children}</div>;
+};
+
+export const Accordion = Object.assign(AccordionRoot, {
```
You don't need to do this. You can bind components to the root component. Example: (https://github.com/oguzergul/compound-component/blob/main/components/Tab/index.js)

ssibrahimbas commented:
## The last screens

### Mobile

![Screenshot 2023-01-26 at 21 39 41](https://user-images.githubusercontent.com/76786120/214922418-33dafdaa-68f8-42d6-aa71-7df6752fa789.png)

### Tablet

![Screenshot 2023-01-26 at 21 39 52](https://user-images.githubusercontent.com/76786120/214922448-3bf572af-5054-484b-a587-f34acf7bd1cf.png)


### Web

![Screenshot 2023-01-26 at 21 39 59](https://user-images.githubusercontent.com/76786120/214922477-d00dc678-cf90-46c3-bd54-1aa7a4497f7c.png)
